httpcore/_async/base.py

```diff
@@ -74,10 +78,10 @@ class AsyncHTTPTransport:
     async def request(
         self,
         method: bytes,
-        url: Tuple[bytes, bytes, int, bytes],
-        headers: List[Tuple[bytes, bytes]] = None,
+        url: URL,
+        headers: Optional[Headers] = None,
```

Comment:
It doesn't seem like it was discussed in #50, but I _think_ there was originally an intentional decision for using explicit types on these datastructures, instead of type aliases.

I'm _personally_ okay with using private aliases, but maybe we should switch the naming to `URLType`, `HeadersType`, etc, to clarify that they're not objects but just aliases to builtin types.

Reply:
So I went with what was discussed on https://github.com/encode/httpx/issues/789, though I guess I should've held off until it was resolved.

It is kinda the point of this PR though, maybe I should just close it?

Reply:
Actually, since `_types` is already in master it's probably better to just move forwards with this model, even if it's only for consistency.